zkevm-circuits/src/keccak_circuit/keccak_packed.rs

```diff
@@ -0,0 +1,1770 @@
+            // lookups println!("{} + {}", input_part.num_bits,
+            // extra_part.num_bits);
+            if range_check {
+                for part in [input_part, extra_part] {
+                    meta.lookup(name, |_| vec![(part.expr.clone(), range_check_table)]);
+                    *lookup_counter += 1;
+                }
+            }
+
+            parts.push(Part {
+                num_bits: target_sizes[counter],
+                expr,
+                parts: vec![input_part.expr.clone(), extra_part.expr.clone()],
+            });
+
+            counter += 1;
+        } else {
+            unreachable!();
+        }
+    }
+    parts
+}
+
+fn combine_sub_parts_value(input: Vec<PartValue>, part_size: usize) -> Vec<PartValue> {
+    //println!("{:?}", input);
+
+    let target_sizes = target_part_sizes(part_size);
+
+    //println!("{:?}", target_sizes);
+
+    let mut counter = 0;
+    let mut parts = Vec::new();
+    let mut input_iter = input.iter();
+    while let Some(input_part) = input_iter.next() {
+        if input_part.num_bits == target_sizes[counter] {
+            parts.push(*input_part);
+            counter += 1;
+        } else if let Some(extra_part) = input_iter.next() {
+            assert_eq!(
+                input_part.num_bits + extra_part.num_bits,
+                target_sizes[counter]
+            );
+
+            let factor = 8u64.pow(input_part.num_bits as u32);
+            let value = input_part.value + extra_part.value * factor;
+
+            parts.push(PartValue {
+                value,
+                num_bits: target_sizes[counter],
+            });
+
+            counter += 1;
+        } else {
+            unreachable!();
+        }
+    }
+    parts
+}
+
+impl<F: Field> KeccakPackedConfig<F> {
+    pub(crate) fn configure(meta: &mut ConstraintSystem<F>, r: F) -> Self {
+        let q_enable = meta.selector();
+        let q_first = meta.selector();
+        let q_round = meta.fixed_column();
+        let q_absorb = meta.fixed_column();
+        let q_round_end = meta.selector();
+        let is_final = meta.advice_column();
+        let hash_rlc = meta.advice_column();
+        let state = array_init::array_init(|_| meta.advice_column());
+        let absorb_from = meta.advice_column();
+        let absorb_data = meta.advice_column();
+        let absorb_result = meta.advice_column();
+        let squeeze_packed = meta.advice_column();
+        let round_cst = meta.fixed_column();
+        let normalize_3 = array_init::array_init(|_| meta.lookup_table_column());
+        let normalize_4 = array_init::array_init(|_| meta.lookup_table_column());
+        let normalize_6 = array_init::array_init(|_| meta.lookup_table_column());
+        let chi_base_table = array_init::array_init(|_| meta.lookup_table_column());
+        let chi_ext_table = array_init::array_init(|_| meta.lookup_table_column());
+        let pack_unpack_table = array_init::array_init(|_| meta.lookup_table_column());
```

Comment:
Less table space -> more lookups required -> more memory needed, I think is generally true. So I think nothing we can really do on the circuit to improve this further (except finding better ways to pack more stuff in fewer lookups of course).

Depending on where the peak memory is it may be easy to improve on the prover side (currently this is not handled very efficiently).